hazelcast/src/test/java/com/hazelcast/map/InterceptorTest.java

```diff
@@ -90,6 +88,44 @@ public void testMapInterceptor() throws InterruptedException {
 
     }
 
+    @Test
+    public void testMapInterceptorOnNewMember() throws InterruptedException {
+        TestHazelcastInstanceFactory nodeFactory = createHazelcastInstanceFactory(2);
+        Config config = getConfig();
+        HazelcastInstance instance1 = nodeFactory.newHazelcastInstance(config);
+        IMap map = instance1.getMap("map");
+        for (int i = 0; i < 100; i++) {
+            map.put(i, i);
+        }
+        map.addInterceptor(new NegativeInterceptor());
```

Comment:
This part of the test (so the next 3 lines) are not the primary purpose of this test. You want to verify that the map interceptor is applied on the new member. But it looks to me that you are verifying if the replication is working when a new member is joining the cluster.

Also this test doesn't following:

setup
do
verrify

You are doing

setup
do
verify
do 
verify

This is a typical example of testing too much. So probably better to split it into 2 tests.